piecrust/src/imports.rs

```diff
@@ -452,15 +457,24 @@ fn owner(fenv: Caller<Env>, mod_id_ofs: usize) -> WasmtimeResult<i32> {
             mod_id
         });
 
+        if env.instance(&mod_id).is_none() {
+            let _ = env.create_instance(mod_id);
+        }
```

Comment:
I can't understand this construction. Could you clarify?

Reply:
Without this construction getting metadata (even including owner) fails before the first call to a contract. For price hint metadata, which we need to know before calling the contract, it is critical to have it available all the time, not only after first call. In general, I think such behaviour is a bug. If you can suggest a different fix for this, I'd be happy to see it.

Reply:
Oh, I understand now. Ideally we wouldn't need to instantiate it to get at the contract. The metadata should be available in the store, not the instance.

Reply:
Problem is that when calling `contract_metadata` we don't know if contract has been loaded, i.e., it exists in the commit, but its metadata might still be missing in ContractSession contracts. One solution would be to do something like:
```
pub fn contract_metadata(
        &mut self,
        contract_id: &ContractId,
    ) -> Option<&ContractMetadata> {
        let _ = self.contract(contract_id.clone());
        self.contracts
            .get(contract_id)
            .map(|store_data| store_data.metadata.data())
    }
```
Yet it has an ugly side that it works on `&mut self` now. Another solution would be to preload all contracts up front, but that could be very wasteful. So I tend to think that the above solution is lesser evil. Basically, we have a caching situation here, hence, an apparently "read-only" function is "read/write". Or we could call env.memory_len(mod_id) instead as a quick solution, although cryptic and surprising for future reader (we could rename memory_len to sth more revealing). What do you think?

Reply:
So in line 461 we could have:
`let _ = env.memory_len(mod_id);`
or, if we copy and rename this method:
`env.load_metadata(mod_id);`

Reply:
After an off-line conversation we decided to go with the above solution.